codec-http2/src/main/java/io/netty/handler/codec/http2/Http2Stream.java

```diff
@@ -44,6 +61,26 @@
     State state();
 
     /**
+     * Get the state as related to the {@link Http2LocalFlowController}.
+     */
+    FlowControlState localFlowState();
+
+    /**
+     * Set the state as related to the {@link Http2LocalFlowController}.
+     */
+    void localFlowState(FlowControlState state);
```

Comment:
So I would like to have the accessors, but not necessarily the mutators on the interface. I'm not too happy with this...any suggestions?  I could re-think the factory pattern...but the current code is taking advantage of changing the object to reduce memory consumption when connection.listener events happen.